### PR TITLE
Fix/snaptoken error

### PR DESCRIPTION
**Pull request by uziwuzzy**
1. Fix crash when using snaptoken on credit card
2. Update secure-preferences version to 0.17
3. Enable demo app to receive snaptoken as an input.

### CODE REVIEW

**Review by askur87 — COMMENTED** [review]
LGTM

File: demo/src/main/java/com/midtrans/demo/DemoOrderReviewActivity.java [hunk 1]
```diff
@@ -55,11 +56,12 @@
     private FancyButton payBtn;
     private Toolbar toolbar;
     private ImageView editCustBtn, saveCustBtn, cancelCustBtn, editDelivBtn;
-    private TextInputEditText editName, editEmail, editPhone;
+    private TextInputEditText editName, editEmail, editPhone, editsnap;
```
Comment (farhanpahlevi) [inline demo/src/main/java/com/midtrans/demo/DemoOrderReviewActivity.java:59]: camel case?

File: demo/src/main/java/com/midtrans/demo/DemoOrderReviewActivity.java [hunk 2]
```diff
@@ -92,6 +94,8 @@ private void bindViews() {
         deliveryAddress = (DemoTextView) findViewById(R.id.delivery_address);
         cityAddress = (DemoTextView) findViewById(R.id.city_address);
         postalCodeAddress = (DemoTextView) findViewById(R.id.zip_address);
+        snapPayBtn = (Button) findViewById(R.id.snaptoken_pay_button);
+        editsnap = (TextInputEditText)findViewById(R.id.edit_snaptoken);
```
Comment (farhanpahlevi) [inline demo/src/main/java/com/midtrans/demo/DemoOrderReviewActivity.java:98]: space after )

**Review by farhanpahlevi — CHANGES_REQUESTED** [review]

**Review by femmerling — APPROVED** [review]
LGTM

**Review by farhanpahlevi — APPROVED** [review]

**Pull request merged**